src/Servant/QuickCheck/Internal/HasGenRequest.hs

```diff
@@ -14,130 +14,138 @@ import Prelude.Compat
 import Servant
 import Servant.API.ContentTypes (AllMimeRender (..))
 import Servant.Client           (BaseUrl (..), Scheme (..))
-import Test.QuickCheck          (Arbitrary (..), Gen, elements, oneof)
+import Test.QuickCheck          (Arbitrary (..), Gen, elements, frequency)
 
 import qualified Data.ByteString as BS
 import qualified Data.ByteString.Internal as BS (c2w)
 
 
+runGenRequest :: HasGenRequest a => Proxy a -> Gen (BaseUrl -> Request)
+runGenRequest = snd . genRequest
+
+
 class HasGenRequest a where
-    genRequest :: Proxy a -> Gen (BaseUrl -> Request)
+    genRequest :: Proxy a -> (Int, Gen (BaseUrl -> Request))
+
 
 instance (HasGenRequest a, HasGenRequest b) => HasGenRequest (a :<|> b) where
     genRequest _
-      = oneof [ genRequest (Proxy :: Proxy a)
-              , genRequest (Proxy :: Proxy b)
-              ]
+      = (lf + rf, frequency [l, r])
+      where
+        l@(lf, _) = genRequest (Proxy :: Proxy a)
+        r@(rf, _) = genRequest (Proxy :: Proxy b)
```

Comment:
I had a test that created a list of requests and checked the paths for an API that has endpoints like this: "/1", "/2", etc. For some reason, the test would only ever return a huge list of "/1". I thought I must have been doing something wrong and I looked at it again and again.

Finally, I looked here and spotted the problem: both lines were returning `genRequest (Proxy :: Proxy a)`. After I swapped `a` for `b`, it worked as expected.